modules/shared/src/main/scala/io/constellationnetwork/security/Hash.scala

```diff
@@ -24,12 +24,42 @@ object hash {
   }
 
   object Hash {
+    private val hexDigits = "0123456789abcdef".toCharArray
+    private val sha256 = MessageDigest.getInstance("SHA-256")
 
+    case class Sha256Digest(private val bytes: Array[Byte]) {
+      require(bytes.length == 32, "invalid SHA-256 digest length")
+
```

Comment:
instead of throwing errors, you should make the case class constructor private and create an apply function for `Sha256Digest` that will explicitly return an Either. in FP, you want to avoid side effects.  another way would be to use refined types if possible

Reply:
Done